### ISSUE TITLE
Single-line CodeSnippet (react-) component does not have an associated visible label

### ISSUE THREAD

**Issue by daniel-schulz-privat** [carbon-design-system/carbon#6516]
## Single-line component does not have an associated visible label
When I run an AVT scan (Chrome DAP) of the single-line code snippet, I get the potential AVT violation "An input element must have an associated visible label".

## Environment

> MacOS

> Chrome

> DAP plugin for Chrome

## Detailed description

> Latest, as I scanned the page https://react.carbondesignsystem.com/?path=/story/codesnippet--single-line

> I do expect the potential violation to be not there

> The potential violation is there

> related to https://www.w3.org/WAI/WCAG21/Techniques/html/H44

## Steps to reproduce the issue

1. go to https://react.carbondesignsystem.com/?path=/story/codesnippet--single-line
2. scan page using DAP for chrome
3. spot the potential violation in the report (note that the storybook uses frames, so you might have to select the correct frame in the DAP report)

> sandbox https://codesandbox.io/s/carbon-components-react-skcbc

Note that with the codesandbox there are iframes involved as well. After you run the DAP scan, please select the frame "React App" to see the report for the contents of the sandbox itself.

## Additional information
With the screenshot below you see that there is a aria-label parameter. This can be set by specifying the parameter ariaLabel with the react component. However, an "aria-labeldby" parameter can not be set.

Having said this, this potential violation could be handled by allowing to specify a parameter to the React-CodeSnippet component that would be used to add a 'aria-labeledby' parameter to the role="textbox" element. This way one could manually specify a visible label outside of this component.

![image](https://user-images.githubusercontent.com/20154051/88185036-baf15d00-cc33-11ea-91d6-cc3e07bf46c2.png)

**Comment by dakahn** [comment]
This is a DAP/AC bug fixed in latest Accessibility Checker preview. [see this comment](https://github.com/carbon-design-system/carbon/issues/6693#issuecomment-682945004)
